projects/plugins/boost/app/features/guide/src/index.ts

```diff
@@ -0,0 +1,42 @@
+import { attachGuides } from './Attach';
+import { createImageObserver } from './ImageObserver';
+import { load } from './Images';
+import { measure } from './Measurements';
+import AdminBarToggle from './ui/AdminBarToggle.svelte';
+
+/**
+ * Initialize the admin bar toggle.
+ */
+document.addEventListener( 'DOMContentLoaded', () => {
+	const adminBarToggle = document.getElementById( 'wp-admin-bar-jetpack-boost-image-guide' );
+	const link = adminBarToggle?.querySelector( 'a' );
+	if ( adminBarToggle && link ) {
+		const href = link.getAttribute( 'href' );
+		link.remove();
+		new AdminBarToggle( {
+			target: adminBarToggle,
+			props: {
+				href,
+			},
+		} );
+	}
+} );
+
+/**
+ * Initialize the guides when window is loaded.
+ */
+window.addEventListener( 'load', async () => {
+	const nodes = document.querySelectorAll( 'body *' );
+	const images = await load( Array.from( nodes ) );
+	const measuredImages = measure( images );
+	attachGuides( measuredImages );
```

Comment:
Not everything has to be completely DRY, some repetition is ok, especially considering that DRYing up here would probably result only in more lines of code.

Reply:
Keeping it DRY in this case would make it easy to trigger an update on toggle of the guide.

Reply:
Are we talking about keeping these 3 lines DRY or the tasks that they perform? Because I'm not seeing an issue copying 3 lines over to another function that listens for clicks.

I'll take a look as I go when adding the toggle listener, I might just put it in another function if it makes sense at that point.

Reply:
I am seeing the attachment of the guides as a task. The task can perform on page load or because of a manual refresh or observed change.

Reply:
Yeah, I don't see why there can't be a method which takes in the value of `nodes` and handles it from there?

Reply:
Yep, I agree. But it's also a task that can be performed with 3 lines of code already. If we'd start de-duping all 3 line code blocks that can be identified as separate tasks we'd run into problems real quick :) 

But I see what you mean - I'll consider it.

Reply:
Turns out this doesn't need to be a DRY function in this case after all 🙂